src/black/__init__.py

```diff
@@ -6632,7 +6634,7 @@ def can_omit_invisible_parens(
             # unnecessary.
             return True
 
-        if line.should_explode and penultimate.type == token.COMMA:
+        if line.magic_trailing_comma and penultimate.type == token.COMMA:
```

Comment:
It may be possible to remove this conditional check block completely, with only a minor impact on the formatter's behaviour.  See https://github.com/jayaddison/black/commit/1096bc908d56a6013c82f626f08f2915a38ecd32 for a possible change (cc @JelleZijlstra)

I'm not including that commit here yet, since I think it'd be good for this refactoring-focused pull request to maintain near-zero change in formatter behaviour.  In practice I think there are small logical changes here, but I hope they're extremely minimal.

Reply:
This commit has been included into #1958, so it can be skipped here.